### PR TITLE
Mouly HW1 AddUtil

### PR DESCRIPTION
**Pull request by chandramouly**
Mouly HW1 AddUtil
Submitted on 11.59pm 2/1

### CODE REVIEW

**Review by manojgo — COMMENTED** [review]
Got the patch. Good going.

File: AddUtil.java [hunk 1]
```diff
@@ -16,10 +16,60 @@ public static void myassert(boolean  x) {
     }
     private static int add(int a, int b){
         //Write your code here
+        return (a + b);
     }
 
     private static String add(String a, String b) {
         //Write your code here
+        if((a.charAt(0) == '-') && (b.charAt(0) == '-')) {
```
Comment (balamuralim) [inline AddUtil.java:24]: Use constant Strings if you use in multiple places '-'

File: AddUtil.java [hunk 2]
```diff
@@ -16,10 +16,60 @@ public static void myassert(boolean  x) {
...
+        private static String addPvt(String a, String b) {
+        // Handle some base cases
+        if(a.length() == 0){
+            return b;
+        }
+        if(b.length() == 0){
+            return a;
+        }
+        StringBuilder sum = new StringBuilder();
+        int aLen = a.length()-1, bLen = b.length()-1;
```
Comment (balamuralim) [inline AddUtil.java:41]: Better to assign a.length() to a variable.

File: AddUtil.java [hunk 3]
```diff
@@ -16,10 +16,60 @@ public static void myassert(boolean  x) {
...
+                aLen--;
+                bLen--;
+            }
+            while(aLen >= 0){
+                tempSum = (a.charAt(aLen) - '0') + remainder;
+                sum.append(tempSum);
+                remainder = 0;
+                aLen--;
+            }
+            while(bLen >= 0){
```
Comment (balamuralim) [inline AddUtil.java:58]: Code duplication (2 while loops). This can be moved to a method **Reusability**

File: AddUtil.java (hunk 2 above)
Comment (chandramouly) [reply AddUtil.java:41]: Yes a.length() is assigned to aLen, sorry missing something?

Comment (balamuralim) [reply AddUtil.java:41]: Line 34 also uses a.length(). Better practice, to assign it to a variable if used at  multiple places

File: AddUtil.java (hunk 3 above)
Comment (chandramouly) [reply AddUtil.java:58]: ya good point, but wondering how I can do it here? any inputs is appreciated.. if i have to write a reusable function, then i need to pass lot of parameters..

File: AddUtil.java (hunk 1 above)
Comment (chandramouly) [reply AddUtil.java:24]: valid point

File: AddUtil.java (hunk 3 above)
Comment (balamuralim) [reply AddUtil.java:58]: sum, remainder are class variables. You dont have to pass them as params.

Comment (balamuralim) [inline AddUtil.java]: `while ((aLen >= 0) && (bLen >= 0)) {
 +                tempSum = (a.charAt(aLen) - '0') + (b.charAt(bLen) - '0') + remainder;
 +                remainder = tempSum / 10;
 +                tempSum = tempSum % 10;
 +                sum.append(tempSum);
 +                aLen--;
 +                bLen--;
 +            }`

**Comment by balamuralim** [comment]
@chandramouly : Please address the code review comments. I will give 🚢 or :shipit:

**Pull request merged**